src/email.c

```diff
@@ -34,66 +34,123 @@ compose_mail_mailto (GAppInfo       *info,
 {
   g_autofree char *enc_subject = NULL;
   g_autofree char *enc_body = NULL;
+  g_autofree char *arg = NULL;
   g_autoptr(GString) url = NULL;
   g_autoptr(GList) uris = NULL;
+  g_autoptr(GString) tb_arg = NULL;
   int i;
   gboolean success;
   const char *sep;
+  const char *is_tb_snap;
+  const char *old_cmdline;
 
   enc_subject = g_uri_escape_string (subject ? subject : "", NULL, FALSE);
   enc_body = g_uri_escape_string (body ? body : "", NULL, FALSE);
 
-  url = g_string_new ("mailto:");
-
-  sep = "?";
-
-  if (addrs)
-    {
-    for (i = 0; addrs[i]; i++)
-      {
-        if (i > 0)
-          g_string_append (url, ",");
-        g_string_append_printf (url, "%s", addrs[i]);
+  /* The commandline can be complicated, e.g.
+   * env BAMF_DESKTOP_FILE_HINT=/var/lib/snapd/.../thunderbird_thunderbird.desktop /snap/bin/thunderbird %u
+   * so just match for thunderbird.
+   */
+  old_cmdline = g_app_info_get_commandline(info);
+  if(strstr(old_cmdline, "thunderbird")) {
+    /* Construct the argument to -compose. */
+    tb_arg = g_string_new("");
+    if (addrs) {
+      g_string_append_printf(tb_arg, "to='");
+      for (i = 0; addrs[i]; i++){
+        g_string_append_printf(tb_arg, "%s,", addrs[i]);
       }
+      g_string_append_printf(tb_arg, "',");
+    }
+    if (cc) {
+      g_string_append_printf(tb_arg, "cc='");
+      for (i = 0; cc[i]; i++){
+        g_string_append_printf(tb_arg, "%s,", cc[i]);
+      }
+      g_string_append_printf(tb_arg, "',");
+    }
+    if (bcc) {
+      g_string_append_printf(tb_arg, "bcc='");
+      for (i = 0; bcc[i]; i++){
+        g_string_append_printf(tb_arg, "%s,", bcc[i]);
+      }
+      g_string_append_printf(tb_arg, "',");
+    }
+    g_string_append_printf (tb_arg, "subject='%s',", enc_subject);
+    g_string_append_printf (tb_arg, "body='%s',", enc_body);
+    if (attachments) {
+      g_string_append_printf(tb_arg, "attachment='");
+      for (i = 0; attachments[i]; i++){
+        if (i != 0) g_string_append_printf (tb_arg, ",");
+        gchar *enc_att = (g_uri_escape_string(attachments[i], "/", 1));
+        g_string_append_printf (tb_arg, "file://%s", enc_att);
+      }
+      g_string_append_printf(tb_arg, "',");
     }
 
-  if (cc)
-    {
-      g_string_append_printf (url, "%scc=", sep);
-      sep = "&";
-
-      for (i = 0; cc[i]; i++)
+    is_tb_snap = strstr(old_cmdline, "/snap/bin/thunderbird");
```

Comment:
Need to check this from the application type

Reply:
Sorry, what is that? Does it have something to do with "content type" as in https://docs.gtk.org/gio/iface.AppInfo.html?